src/ConfigCatContext.tsx

```diff
@@ -14,4 +14,23 @@ const ConfigCatContext = React.createContext<ConfigCatContextData | undefined>(
 
 ConfigCatContext.displayName = "ConfigCatContext";
 
+const ConfigCatContextRegistry = new Map<string, React.Context<ConfigCatContextData | undefined>>();
```

Comment:
Another remark: it slightly bothers me that we only add items to this map but never remove. Of course, it's not clear when we could remove unused items...

However, this variable is defined on the module level, which means it will live during the whole lifetime of the application, that is, what is put in the map will also be kept alive until the application terminates. This looks like a soft memory leak. 

Maybe storing [weak references](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakRef) instead would solve this, but I can't tell without deeper knowledge of the framework.

Apart from extreme cases, this probably won't cause a big problem, so the current implementation is fine by me, it just doesn't feel very good. :)

Reply:
It depends on the consumer, with this it is possible to create many items with randomly providerIds.
I can not see how is it possible to use weakRef because there is no direct connection between contextAPI (and its usage) and our dictionary. If a developer creates the provider and it won't use it I think we can not do anything :/

Reply:
> I can not see how is it possible to use weakRef because there is no direct connection between contextAPI (and its usage) 

Something like this: `new Map<string, WeakRef<React.Context<ConfigCatContextData | undefined>>>`. But it's not a path I'd like to go down, that's for sure..

> If a developer creates the provider and it won't use it I think we can not do anything :/

What seems really problematic is that providers can even be created on the "consumer side", e.g. in "useFeatureFlag", since that function also use `getOrCreateConfigCatContext` internally.

This can be mitigated though if we just _get_ in those cases, but not _create_.

This would improve the situation a bit but wouldn't solve the issue. I'll be thinking about this and see if I can come up with something.